### DISCUSSION TITLE
Docs: Better explain risks and strategies for including large local images.

### DISCUSSION THREAD

**Discussion by k-funk** [vercel/next.js#62617] [Help]
### What is the improvement or update you wish to see?

https://image-component.nextjs.gallery/ is a nice start to image docs, but there's a lot more that needs to be covered to reduce the steep learning curve required to not shoot yourself in the foot, which is easy with the `<Image />` component.

A few questions that I have, that docs could help with:
* How do you recommend handling large images? Should I be doing some downsizing/optimizing before putting them in `/public`, or can I trust NextJS? right now, using this component is actually quite worrisome for me.
* Code examples for an image galleries of different styles (masonry, percent-based widths, using `<Image layout="fill" objectFit="cover" />`, other `layout` and `objectFit` usages)
* Code examples of using `sizes` with explanations based on the images usages. Also demonstrate the risk of *not* using it. Possibly a warning banner, since in my case, it seems *required* to have reasonable asset-loading times
* Code examples for usages, ie: `position: 'relative'` seems like a commonly asked stackoverflow question, as it's not an intuitive requirement
* Can I define an image based on a single-dimension? ie: not more than 300px wide? (whenever i try this, the image usually gets stretched), I'd rather do that as my default, rather than relying on me getting the `sizes` string right

### Is there any context that might help us understand?

I find the `<Image />` to be a steep learning curve. For example, I've not used `srcset` or `sizes` before. Even debugging them is a bit of a challenge:
* in order to test what users will actually see, needing to either disable cache/use an incognito window. hard refresh doesn't work
* inspecting the `properties` tab of the style panel to see what the `currentSrc` is
* guessing at what my browser is actually try to load when i tell is `1000px`, based on my device's pixel density

For my designs, I'm using a lot of `<Image layout="fill" objectFit="cover" />`, filling a bootstrap column-based layout (percentages within a fixed or fluid-width div). I've included images in my `public` directory that are large 1-2mb jpegs. However, what I'm finding is that by default, the browser is typically loading 3840px or 2048px-wide images, when the rendered size is at most 360x360 in my usage. There is some free image performance, as this will cut them down to ~300kb-800kb, but for an image gallery of 30 images, ~18mb is pretty significant for out of the box behavior.

<img width="1644" alt="Screen Shot 2022-06-05 at 22 34 30" src="https://user-images.githubusercontent.com/1933021/172268474-8395a0a4-1e7f-46f1-9c5b-18dca0c796f0.png">

Also, even though the [docs say that "Next.js supports IE11 and all modern browsers (Edge, Firefox, Chrome, Safari, Opera, et al) with no required configuration"](https://nextjs.org/docs/basic-features/supported-browsers-features), `srcset` and `sizes` [only started being supported in in Chrome in late 2014](https://caniuse.com/srcset). So if a browser doesn't support them, the user is going to get the 3048px images.

I'm actually considering going back to using `<img />` and doing pre-processing myself (generating thumbnails per size, the old-school way) just because I'm not confident in my usage of this component that promises so much but is quite dangerous in practice.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/api-reference/next/image

**Comment by samcx** [comment]
Converting this to discussions since this is a :nextjs: related question.
